pkg/plugins/resources/postgres/pgx_store.go

```diff
@@ -183,15 +187,42 @@ func (r *pgxResourceStore) List(ctx context.Context, resources core_model.Resour
 	var statementArgs []interface{}
 	statementArgs = append(statementArgs, resources.GetItemType())
 	argsIndex := 1
-	if opts.Mesh != "" {
-		argsIndex++
-		statement += fmt.Sprintf(" AND mesh=$%d", argsIndex)
-		statementArgs = append(statementArgs, opts.Mesh)
-	}
-	if opts.NameContains != "" {
-		argsIndex++
-		statement += fmt.Sprintf(" AND name LIKE $%d", argsIndex)
-		statementArgs = append(statementArgs, "%"+opts.NameContains+"%")
+	rkSize := len(opts.ResourceKeys)
+	if rkSize > 0 && rkSize < int(r.maxListQueryElements) {
+		statement += " AND ("
+		res := resourceNamesByMesh(opts.ResourceKeys)
+		iter := 0
+		for mesh, names := range res {
+			if iter > 0 {
+				statement += " OR "
+			}
+			argsIndex++
+			statement += fmt.Sprintf("(mesh=$%d AND", argsIndex)
+			statementArgs = append(statementArgs, mesh)
+			for idx, name := range names {
+				argsIndex++
+				if idx == 0 {
+					statement += fmt.Sprintf(" name IN ($%d", argsIndex)
+				} else {
+					statement += fmt.Sprintf(",$%d", argsIndex)
+				}
+				statementArgs = append(statementArgs, name)
+			}
+			statement += "))"
+			iter++
+		}
+		statement += ")"
+	} else {
```

Comment:
I think this should be outside of else, no?

Reply:
I am not sure if that makes sense. ResourceKey requires name and mesh to be set. It would filter this twice.

Reply:
It might not make huge sense to create that query but it can be possible